dockerfiles/cli/scripts/cli.sh

```diff
@@ -115,37 +114,40 @@ cmd_init_reinit_pre_action() {
 cmd_start_check_ports() {
 
   # If dev mode is on, then we also need to check the debug port set by the user for availability
-  if [ "${CHE_DEVELOPMENT_MODE}" = "on" ]; then
+  if debug_server; then
     USER_DEBUG_PORT=$(docker_run --env-file="${REFERENCE_CONTAINER_ENVIRONMENT_FILE}" alpine sh -c 'echo $CODENVY_DEBUG_PORT')
 
     if [[ "$USER_DEBUG_PORT" = "" ]]; then
       # If the user has not set a debug port, then use the default
       CODENVY_DEBUG_PORT=8000
+      CHE_DEBUG_PORT=8000
     else 
       # Otherwise, this is the value set by the user
       CODENVY_DEBUG_PORT=$USER_DEBUG_PORT
+      CHE_DEBUG_PORT=$USER_DEBUG_PORT
     fi
   fi
 
   text   "         port 80 (http):        $(port_open 80 && echo "${GREEN}[AVAILABLE]${NC}" || echo "${RED}[ALREADY IN USE]${NC}") \n"
   text   "         port 443 (https):      $(port_open 443 && echo "${GREEN}[AVAILABLE]${NC}" || echo "${RED}[ALREADY IN USE]${NC}") \n"
   text   "         port 5000 (registry):  $(port_open 5000 && echo "${GREEN}[AVAILABLE]${NC}" || echo "${RED}[ALREADY IN USE]${NC}") \n"
   text   "         port 2181 (zookeeper): $(port_open 2181 && echo "${GREEN}[AVAILABLE]${NC}" || echo "${RED}[ALREADY IN USE]${NC}") \n"
-  if [ "${CHE_DEVELOPMENT_MODE}" = "on" ]; then
+  if debug_server; then
     text   "         port ${CODENVY_DEBUG_PORT} (debug):     $(port_open ${CODENVY_DEBUG_PORT} && echo "${GREEN}[AVAILABLE]${NC}" || echo "${RED}[ALREADY IN USE]${NC}") \n"
     text   "         port 9000 (lighttpd):  $(port_open 9000 && echo "${GREEN}[AVAILABLE]${NC}" || echo "${RED}[ALREADY IN USE]${NC}") \n"
   fi
 
-  if ! $(port_open 80) || ! $(port_open 443) || ! $(port_open 5000); then
+  if ! $(port_open 80) || ! $(port_open 443) || ! $(port_open 5000) || ! $(port_open 2181); then
     echo ""
     error "Ports required to run $CHE_MINI_PRODUCT_NAME are used by another program."
-    return 1;
+    return 2;
   fi
-  if [ "${CHE_DEVELOPMENT_MODE}" = "on" ]; then
-    if ! $(port_open ${CODENVY_DEBUG_PORT}); then
+
+  if debug_server; then
+    if ! $(port_open ${CODENVY_DEBUG_PORT}) || ! $(port_open 9000); then
       echo ""
       error "Ports required to run $CHE_MINI_PRODUCT_NAME are used by another program."
-      return 1;
+      return 12
```

Comment:
Yes - I am cleaning up the port-check logic to be faster as we now need a lot more checks on it.